### PR TITLE
fix spacing in query-loop and post-title size in single/post templates

### LINKED ISSUE #190: Fix spacing in Home template's query loop
Currently, the home template's query loop spacing is looking way too large. This should be matched with the intended design:

| Local site      | Intended design |
| ----------- | ----------- |
|   Captura de ecrã 2022-09-21, às 16 51 15  |  Captura de ecrã 2022-09-21, às 16 54 29    |

### PR DESCRIPTION
**Pull request by madhusudhand**
This change fixes

* spacing around post title in homepage query loop. 
* adjusts the post-title to `h4` in query loop
* adjusts post title to `h2` in single and post templates.

| Before | After |
| --- | --- |
| image | image |

Fixes: #190

### CODE REVIEW

**Comment by madhusudhand** [comment]
> adjusts the post-title to h4 in query loop

@mikachan By referring designs h4 is the right size, but this changes seems to reverse the change done in #111. 
I might be missing the context, can you confirm if it was intentional?

**Comment by mikachan** [comment]
> By referring designs h4 is the right size, but this changes seems to reverse the change done in https://github.com/WordPress/twentytwentythree/pull/111.

These titles were changed from h4 to h3 to maintain the correct heading hierarchy, as the 'Latest Posts' h3 heading was [removed here](https://github.com/WordPress/twentytwentythree/pull/111/files#diff-6d73f6bbcc50527a0567c49995acacf9cdf1e28ea880751d352a5a914b87e93dL19). 

We shouldn't really be changing the heading level just for styling reasons. Perhaps we need to reduce the font size of these headings via the query block settings instead?

File: templates/home.html [hunk 1]
```diff
@@ -15,7 +15,7 @@ <h2 class="alignwide" style="margin-top:var(--wp--preset--spacing--60);margin-bo
 	<div class="wp-block-query alignwide">
 		<!-- wp:post-template {"align":"wide"} -->
 			<!-- wp:post-featured-image {"isLink":true,"width":"100%","height":"max(15vw, 30vh)","align":"wide"} /-->
-			<!-- wp:post-title {"level":3,"isLink":true} /-->
+			<!-- wp:post-title {"level":4,"isLink":true} /-->
```
Comment (mikachan) [inline templates/home.html]: I think this should stay as a h3, based on [this comment](https://github.com/WordPress/twentytwentythree/pull/191#issuecomment-1254887001).

**Review by mikachan — COMMENTED** [review]
Thanks for this! The spacing change on the post title block looks good.

This PR has made me reconsider how we're using heading tags across the site. After looking at each template, I don't think we should be using a h1 for the site title in the header (it should just be a paragraph), and instead, ensure that each template has a contextual h1 (e.g. the query title on the archive page is a h1). What do you think?

**Comment by madhusudhand** [comment]
> I don't think we should be using a h1 for the site title in the header (it should just be a paragraph), and instead, ensure that each template has a contextual h1

I Agree. It isn't contextually right to use h1 for Site title, it can just be a paragraph and bold font to be given by its class, where as for H1, the archive page title gives the right context.

File: templates/page.html [hunk 2]
```diff
@@ -4,7 +4,7 @@
 <main class="wp-block-group">
 	<!-- wp:group {"layout":{"type":"constrained"}} -->
 	<div class="wp-block-group">
-		<!-- wp:post-title {"level":1,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
+		<!-- wp:post-title {"level":2,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
```
Comment (mikachan) [inline templates/page.html]:
```suggestion
		<!-- wp:post-title {"level":1,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
```

If we're going to make the site title a paragraph, I think these headings should stay as h1s.

File: templates/single.html [hunk 3]
```diff
@@ -4,7 +4,7 @@
 <main class="wp-block-group">
 	<!-- wp:group {"layout":{"type":"constrained"}} -->
 	<div class="wp-block-group">
-		<!-- wp:post-title {"level":1,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
+		<!-- wp:post-title {"level":2,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
```
Comment (mikachan) [inline templates/single.html]:
```suggestion
		<!-- wp:post-title {"level":1,"align":"wide","style":{"spacing":{"margin":{"top":"var:preset|spacing|70","bottom":"var:preset|spacing|70"}}}} /-->
```

**Review by mikachan — APPROVED** [review]

**Comment by carolinan** [comment]
1+ for only having one H1 on each page. The home page also needs an H1 though.

**Pull request merged**